components/script/dom/htmllinkelement.rs

```diff
@@ -79,11 +79,16 @@ impl HTMLLinkElement {
     }
 
     pub fn set_stylesheet(&self, s: Arc<Stylesheet>) {
-        assert!(self.stylesheet.borrow().is_none());
-        *self.stylesheet.borrow_mut() = Some(s);
+        let mut sheet = self.stylesheet.borrow_mut();
+        let must_invalidate = sheet.is_some();
```

Comment:
I think we don't need to call `invalidate_stylesheets` here, since the stylesheet loader will do it right after calling `set_stylesheet`.

Remove this code and maybe add a note about it?